cockatrice/src/SpoilerBackgroundUpdaterThread.cpp

```diff
@@ -0,0 +1,233 @@
+{
+    // Determine if we're in spoilers season or not. If we're not, just kill this thread
+    downloadSpoilerSeasonStatusFile();
+    if (! mbIsActiveSpoilerSeason)
+    {
+        qDebug() << "Spoiler Timer disabled. NOT Spoilers Season";
+        return;
+    }
+
+    // Determine if the thread should be active in downloading/updating spoilers
+    mbIsActiveThread = settingsCache->getDownloadSpoilersStatus();
+    connect(settingsCache, SIGNAL(downloadSpoilerStatusChanged()), this, SLOT(changeActiveStatus()));
+
+    mpTimerForSpoilers = new QTimer(this);
+
+    // When timer ends, cause download of Spoilers XML file
+    connect(mpTimerForSpoilers, SIGNAL(timeout()), this, SLOT(timeoutOccurredTimeToDownloadSpoilers()));
+
+    // If there is a change in the settings, update the timer so we know when to update correctly
+    connect(settingsCache, SIGNAL(downloadSpoilerTimeIndexChanged()), this, SLOT(handleNewTimeInterval()));
+
+    // Start the timer sequence and the domino effects
+    runTimer();
+}
+
+SpoilerBackgroundUpdaterThread::~SpoilerBackgroundUpdaterThread()
+{
+    delete mpTimerForSpoilers;
+}
+
+void SpoilerBackgroundUpdaterThread::changeActiveStatus()
+{
+    mbIsActiveThread = (settingsCache->getDownloadSpoilersStatus());
+    qDebug() << "Spoiler Timer running status has been changed to" << mbIsActiveThread;
+}
+
+void SpoilerBackgroundUpdaterThread::timeoutOccurredTimeToDownloadSpoilers()
+{
+    qDebug() << "Spoiler Timer has finished";
+
+    settingsCache->setDownloadSpoilerLastUpdateTime(QDateTime::currentMSecsSinceEpoch());
+
+    // If this thread is active, then initiate download
+    if (mbIsActiveThread)
+    {
+        qDebug() << "Timer thread is active, so attempting to download spoilers";
+        downloadSpoilersFile(QString(GITHUB_SPOILERS_DOWNLOAD_URL));
+    }
+
+    // Now that we've downloaded the spoiler, reset the timer.
+    // Take it from the top, Jenkins!
+    runTimer(true);
+}
+
+void SpoilerBackgroundUpdaterThread::handleNewTimeInterval()
+{
+    // Cause the timer to stop then restart w/ new interval
+    qDebug() << "Spoiler Timer has new time interval established";
+    runTimer(true);
+}
+
+void SpoilerBackgroundUpdaterThread::runTimer(bool lbStopAndRestart)
+{
+    // Cause a restart of the timer as if we just restart via start() a timeout() will be
+    // thrown which will trigger a download... which we don't want off time!
+    if (lbStopAndRestart)
+    {
+        mpTimerForSpoilers->stop();
+    }
+
+    // Determine how long to wait between updates (Gets the # of hours from settingsCache / convert Mins -> Millis)
+    float lnTimeToWaitBetween = settingsCache->getDownloadSpoilerTimeMinutes() * 60000;
+
+    // How much time is needed until the next update (NOW() - Last Update Time == Time Remaining)
+    long lnLastUpdateTime = settingsCache->getDownloadSpoilerLastUpdateTime();
+    long lnTimeUntilNextUpdate = QDateTime::currentMSecsSinceEpoch() - lnLastUpdateTime;
```

Comment:
As far as i understand, `lnTimeUntilNextUpdate` should ne named "lnTimeSinceLastUpdate"